File: RepoTransBench/target_projects/Java/Go/sockeqwe_fragmentargs/tests/original/fragmentargs_test.go
package original

import (
	"reflect"
	"testing"
)

type FragmentArgsInjector interface {
	Inject(target interface{})
}

type FragmentArgs struct{}

var autoMappingInjector FragmentArgsInjector

func (FragmentArgs) inject(target interface{}) {
	if autoMappingInjector != nil {
		autoMappingInjector.Inject(target)
	}
}

func (FragmentArgs) injectFromBundle(target interface{}) {
	// Simulate autoMappingInjector lookup that may fail
	// For coverage: leave as is
}

func TestInjectWithNoAutoMappingClass(t *testing.T) {
	// Should not panic even if injector cannot be found.
	fa := FragmentArgs{}
	fa.inject(struct{}{})
}

type dummyInjector struct {
	injected bool
}

func (d *dummyInjector) Inject(target interface{}) {
	d.injected = true
}

func TestInjectWithAutoMappingInjectorPresent(t *testing.T) {
	// Reflection hack: simulate static field set
	var dummy = &dummyInjector{}
	autoMappingInjector = dummy

	fragment := struct{}{}
	FragmentArgs{}.inject(fragment)
	if !dummy.injected {
		t.Errorf("inject did not call DummyInjector.Inject")
	}

	// Clean up for other tests
	autoMappingInjector = nil
}

File: RepoTransBench/target_projects/Java/C#/sockeqwe_fragmentargs/tests/original/FragmentArgsTest.cs
using System;
using System.Reflection;
using Xunit;

namespace ProjectName.Tests.Original
{
    public class FragmentArgsTest
    {
        // Dummy interface from the library under test
        public interface IFragmentArgsInjector
        {
            void Inject(object target);
        }

        public class DummyInjector : IFragmentArgsInjector
        {
            public bool Injected = false;
            public void Inject(object target)
            {
                Injected = true;
            }
        }

        // Simulate static field in FragmentArgs class
        public static class FragmentArgs
        {
            public static IFragmentArgsInjector? autoMappingInjector { get; set; }
            public static void Inject(object target)
            {
                autoMappingInjector?.Inject(target);
            }
        }

        [Fact]
        public void TestInjectWithNoAutoMappingClass()
        {
            // Should not throw even if injector cannot be found.
            FragmentArgs.autoMappingInjector = null;
            FragmentArgs.Inject(new object());
        }

        [Fact]
        public void TestInjectWithAutoMappingInjectorPresent()
        {
            // Reflection hack to inject DummyInjector (simulate static field).
            var dummy = new DummyInjector();
            FragmentArgs.autoMappingInjector = dummy;

            object fragment = new object();
            FragmentArgs.Inject(fragment);
            Assert.True(dummy.Injected);

            // Clean up for other tests.
            FragmentArgs.autoMappingInjector = null;
        }
    }
}